### PR TITLE
CSPL-444-Added util to get rf sf status from cm

### PR DESCRIPTION
**Pull request by pdhanoyasplunk**

### CODE REVIEW

File: test/testenv/cmutil.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// ClusterMasterHealthResponse is a representation of the health response by a Splunk cluster-master
+// Endpoint: /services/cluster/master/health
+type ClusterMasterHealthResponse struct {
```
Comment (gaurav-splunk) [inline test/testenv/cmutil.go:11]: do we actually need this?I don't see multiple entries in the CM health response. Maybe just rename `ClusterMasterHealthEntry` to `ClusterMasterHealthResponse` and use it directly like `restResponse.Content.SearchFactorMet` ?

File: test/smoke/smoke_test.go [hunk 2]
```diff
@@ -309,6 +323,15 @@ var _ = Describe("Smoke test", func() {
...
 			}, deployment.GetTimeout(), PollInterval).Should(Equal(siteIndexerMap))
 
+			// Verify MC Pod is Ready
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Verify RF SF is met
+			Eventually(func() bool {
+				rfSfStatus := testenv.CheckRFSF(testenvInstance.GetName(), deployment.GetName())
+				testenvInstance.Log.Info("Verifying RF SF is met", "Status", rfSfStatus)
+				return rfSfStatus
```
Comment (sgontla) [inline test/smoke/smoke_test.go:333]: Looks like this code is repetition in all the above 3 places. Wondering if this function can be defined explicitly and called from `Eventually()`?

File: test/testenv/cmutil.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+type ClusterMasterHealthContent struct {
+	AllDataIsSearchable      string `json:"all_data_is_searchable"`
+	AllPeersAreUp            string `json:"all_peers_are_up"`
+	Multisite                string `json:"multisite"`
+	NoFixupTasksInProgress   string `json:"no_fixup_tasks_in_progress"`
+	ReplicationFactorMet     string `json:"replication_factor_met"`
+	SearchFactorMet          string `json:"search_factor_met"`
+	SiteReplicationFactorMet string `json:"site_replication_factor_met"`
+	SiteSearchFactorMet      string `json:"site_search_factor_met"`
+}
```
Comment (sgontla) [inline test/testenv/cmutil.go:31]: Aren't these be best represented with `bool` than `string`

Comment (pdhanoyasplunk) [reply test/testenv/cmutil.go:31]: @sgontla the return from Splunk API is string.

File: test/testenv/cmutil.go (hunk 1 above)
Comment (pdhanoyasplunk) [reply test/testenv/cmutil.go:11]: @gaurav-splunk I kept the hierarchical order of API return for simplicity. In future I am thinking of adding more verifications in future therefore filtering more fields.

File: test/testenv/cmutil.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	podName := fmt.Sprintf("splunk-%s-cluster-master-0", deployment.GetName())
+	stdin := "curl -ks -u admin:$(cat /mnt/splunk-secrets/password) https://localhost:8089/services/cluster/master/health?output_mode=json"
+	command := []string{"/bin/sh"}
+	stdout, stderr, err := deployment.PodExecCommand(podName, command, stdin, false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to execute command on pod", "pod", podName, "command", command)
+		return false
+	}
+	logf.Log.Info("Command executed on pod", "pod", podName, "command", command, "stdin", stdin, "stdout", stdout, "stderr", stderr)
+	restResponse := ClusterMasterHealthResponse{}
```
Comment (kashok-splunk) [inline test/testenv/cmutil.go:45]: Should we invest time in creating an HTTP client for the test suite also? Till then we can have a common API that can exec on the Pod get the json response, unmarshall, and return an interface with all the desired entries from the json. I see the same being done at multiple places Example: https://github.com/splunk/splunk-operator/blob/develop/test/testenv/search_utils.go#L89 / https://github.com/splunk/splunk-operator/blob/develop/test/ingest_search/ingest_search_test.go#L61

Comment (pdhanoyasplunk) [reply test/testenv/cmutil.go:45]: @kashok-splunk sure we can take that up as a new task in upcoming sprint.

**Review by sgontla — APPROVED** [review]

**Review by kashok-splunk — APPROVED** [review]
@pdhanoyasplunk Sure.

**Review by smohan-splunk — APPROVED** [review]

**Pull request merged**